### PR TITLE
Add Total Variation (TV) to metrax

### PR DESCRIPTION
**Pull request by Heidar-An**
## [Issue #63]

Add total variation as defined by here: [Tensorflow Total Variation](https://www.tensorflow.org/api_docs/python/tf/image/total_variation)
Add tests comparing results against Tensorflow

### CODE REVIEW

File: src/metrax/image_metrics_test.py [hunk 1]
```diff
@@ -553,6 +574,60 @@ def test_dice(self, y_true, y_pred):
...
+      ),
+  )
+  def test_total_variation_against_tensorflow(
+      self,
+      images_np: np.ndarray,
+      is_batch: bool,
+  ):
+    """Test that TotalVariation metric computes values close to tf.image.total_variation."""
+    # Calculate TV using Metrax
+    metrax_tv = metrax.TotalVariation.from_model_output(
```
Comment (jshin1394) [inline src/metrax/image_metrics_test.py:611]: could you add a case where you merge this.metrax_tv with other.metrax_tv then call compute()? We want to see that multiple metrics of the same kind can be merged correctly. You can follow this example(https://github.com/google/metrax/blob/main/src/metrax/classification_metrics_test.py#L132-L139).

File: src/metrax/image_metrics_test.py [hunk 2]
```diff
@@ -553,6 +574,60 @@ def test_dice(self, y_true, y_pred):
 
     np.testing.assert_allclose(result, expected, rtol=1e-5, atol=1e-5)
 
+  @parameterized.named_parameters(
+      (
+          'tv_single_channel_3d',
+          TV_IMG_1,
+          False,  # is_batch
```
Comment (jshin1394) [inline src/metrax/image_metrics_test.py:581]: nit: we can remove this comment.

File: src/metrax/image_metrics.py [hunk 3]
```diff
@@ -666,3 +666,69 @@ def compute(self) -> jax.Array:
     """Returns the final Dice coefficient."""
     epsilon = 1e-7
     return (2.0 * self.intersection) / (self.sum_pred + self.sum_true + epsilon)
+
+@flax.struct.dataclass
+class TotalVariation(base.Average):
```
Comment (jshin1394) [inline src/metrax/image_metrics.py:671]: could you add a test case to metrax_test.py testing whether if this metric is jittable or not?

**Review by jshin1394 — COMMENTED** [review]
Thank you so much!

**Comment by rni418** [comment]
> Also, I'm not sure if this is just on my machine (WSL, Ryzen 7 3800H, RTX 3070 Mobile) but some tests tests_psnr... and tests_ssim... and more seem to be failing for me. I reverted to ~HEAD and still found them failing. Should I make an issue about this?

Yes please file an issue. Could you also include the test output as well when filing the issue so it'll help us debug? Thanks!
